### PR TITLE
Remove binder from transpileDeclaration implementation.

### PR DESCRIPTION
**Pull request by dragomirtitian**
Addresses feedback about duplication of the binder.

### CODE REVIEW

File: tests/cases/compiler/isolatedDeclarationErrorsExpandoFunctions.ts [hunk 1]
```diff
@@ -0,0 +1,15 @@
+// @declaration: true
+// @isolatedDeclarations: true
+// @declarationMap: false
+// @target: ESNext
+// @isolatedDeclarationFixedDiffReason: Function declarations are not fixed.
```
Comment (h-joo) [inline tests/cases/compiler/isolatedDeclarationErrorsExpandoFunctions.ts]: I guess better to mention "Expando* functions "

File: tests/baselines/reference/isolated-declarations/auto-fixed/diff/isolatedDeclarationErrorsClasses.d.ts.diff
Comment (h-joo) [inline tests/baselines/reference/isolated-declarations/auto-fixed/diff/isolatedDeclarationErrorsClasses.d.ts.diff:1]: I guess the changes in this test file is a side effect of having the "vanilla" symbol info?

File: src/compiler/transformers/declarations/emitResolver.ts [hunk 2]
```diff
@@ -393,3 +614,50 @@ export function createEmitDeclarationResolver(file: SourceFile): CoreEmitResolve
...
+
+function getEntityNameComponent(identifier: MemberName): __String {
+    const name = getPropertyNameForPropertyNameNode(identifier);
+    if (name && name.indexOf(".")) {
+        return name.replace(/\./g, "..") as __String;
+    }
+    return name;
+}
+
+function getDynamicSymbolName(name: ElementAccessExpression | PropertyName) {
```
Comment (DanielRosenwasser) [inline src/compiler/transformers/declarations/emitResolver.ts:627]: I wonder if you can use `tryGetPropertyAccessOrIdentifierToString` or `entityNameToString`.

**Review by h-joo — APPROVED** [review]

File: src/compiler/transformers/declarations/emitResolver.ts (hunk 2 above)
Comment (dragomirtitian) [reply src/compiler/transformers/declarations/emitResolver.ts:627]: Switched over to `entityNameToString`

**Pull request merged**